x/epoching/keeper/epoch_msg_queue.go

```diff
@@ -341,3 +341,70 @@ func cacheTxContext(ctx sdk.Context, txid []byte, msgid []byte, height uint64) (
 
 	return ctx.WithMultiStore(msCache), msCache
 }
+
+// ClearEpochMsgs clears all processed messages and queue metadata for a given epoch
+func (k Keeper) ClearEpochMsgs(ctx context.Context, epochNumber uint64) error {
+	// Get queue length first for logging
+	if epochNumber == 0 {
+		return errorsmod.Wrap(types.ErrInvalidEpoch, "epoch number cannot be zero")
+	}
+
+	currentEpoch := k.GetEpoch(ctx).EpochNumber
+	if epochNumber >= currentEpoch {
+		return errorsmod.Wrapf(types.ErrInvalidEpoch,
+			"cannot clear current or future epoch %d (current: %d)", epochNumber, currentEpoch)
+	}
+
+	queueLength := k.GetQueueLength(ctx, epochNumber)
+
+	if queueLength == 0 {
+		// No messages to clear
+		return nil
+	}
+
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+
+	// Clear all messages in the queue store
+	store := k.msgQueueStore(ctx, epochNumber)
+
+	// Iterate and delete all messages
+	iterator := storetypes.KVStorePrefixIterator(store, nil)
+	defer func() {
+		if err := iterator.Close(); err != nil {
+			sdkCtx.Logger().Error("Failed to close iterator", "error", err)
+		}
+	}()
+
+	keysToDelete := [][]byte{}
+	for ; iterator.Valid(); iterator.Next() {
+		keysToDelete = append(keysToDelete, iterator.Key())
+	}
+
+	// Delete all message entries
+	for _, key := range keysToDelete {
+		store.Delete(key)
+	}
```

Comment:
How about adding a simple Delete function on `keeper/epoch_msg_queue.go`
Please refer to [this implementation](https://github.com/cosmos/cosmos-sdk/blob/7e5a31f310d1a7079c3e8c2d1156c9e5565961e8/x/distribution/keeper/store.go#L185-L193) on sdk 

We don't need to keep `keysToDelete` list IMO